packages/typespec-ts/test/integration/generated/sharedRoute/src/responses.ts

```diff
@@ -2,18 +2,27 @@
 // Licensed under the MIT license.
 
 import { HttpResponse } from "@azure-rest/core-client";
-import { ResourceOutput } from "./outputModels.js";
+import { ErrorModelOutput, ResourceOutput } from "./outputModels.js";
+
+/** There is no content to send for this request, but the headers may be useful. */
+export interface ListBySubscription204Response extends HttpResponse {
+  status: "204";
```

Comment:
why is the response code 204 for list operation? and is it really expected to have no body in list response ?

Reply:
okay i just created an test case for different status code would return different body shape.

I change the code to 202 now.